phases/03-deep-learning-core/10-mini-framework/code/main.py

```
@@ -0,0 +1,603 @@
+
+
+class Dropout(Module):
+    def __init__(self, p=0.5):
+        super().__init__()
+        self.p = p
+        self.mask = None
+
+    def forward(self, x):
+        if not self.training:
+            return x
+        self.mask = [0.0 if random.random() < self.p else 1.0 / (1 - self.p) for _ in x]
```

Ruff (0.15.7)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)